core/src/test/java/com/linecorp/armeria/server/ServiceRequestContextTest.java

```diff
@@ -167,6 +171,38 @@ void pushWithOldIrrelevantServiceCtx() {
         assertCurrentCtx(null);
     }
 
+    @Test
+    void queryParams() {
+        final QueryParams queryParams = QueryParams.of("param1", "value1",
```

Comment:
I couldn't find the relevance between `queryParams` and `ctx.queryParams()` before I look into `serviceRequestContextWithPathAndQuery`. (I also had to carefully take a look at the parameter if there's something different.)

I think it's better to pass the `queryParams` to the method to create the `ctx` or
we can just inline the `serviceRequestContextWithPathAndQuery()` method and merge those three test cases added. WDYT?

Reply:
> we can just inline the serviceRequestContextWithPathAndQuery() method and merge those three test cases added. WDYT?

I think this way looks much better! 👍  I'll update it!